.github/workflows/notebooks-release.yaml

```diff
@@ -0,0 +1,196 @@
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Release tag for the notebooks release'
+        required: true
+        default: 'v1.32.0'
+        type: string
+      release_name:
+        description: 'Name of the release'
+        required: true
+        default: '2025a'
+        type: string
+      buildconfigs_version:
+        description: 'Version to update the BuildConfigs to (if applicable)'
+        required: false
+        type: string
+      update_buildconfigs:
+        description: 'Update BuildConfigs for CUDA RStudio and RStudio'
+        required: true
+        default: 'true'
+        type: boolean
+      branch:
+        description: "Optional: Provide branch name"
+        required: false
+        type: string
+      user-hash:
+        description: "Optional: Specify a Git hash (it should exists on the branch history)"
+        required: false
+        type: string
+
+env:
+  RELEASE_TAG: ${{ github.event.inputs.tag }}
+  RELEASE_NAME: ${{ github.event.inputs.release_name }}
+  BRANCH: ${{ github.event.inputs.branch }}
+  USER_HASH: ${{ github.event.inputs.user-hash }}
+  REPO_OWNER: opendatahub-io
+  REPO_NAME: notebooks
+  VERSION: ${{ github.event.inputs.buildconfigs_version}}
+
+jobs:
+  #1. Update the params.env and commit.env files with new SHAs
+  Update-manifests:
+    uses: opendatahub-io/notebooks/.github/workflows/notebooks-digest-updater.yaml@main
+    with:
+      branch: ${{ github.event.inputs.branch }}
+      user-hash: ${{ github.event.inputs.user_hash }}
+
+  # 2. Check if the Manifest PR is merged      
+  Manifests-merged:
+    needs: Update-manifests
+    runs-on: ubuntu-latest
+    outputs:
+      pr_merged_m: ${{ steps.check_pr.outputs.pr_merged_m }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check if PR is merged
+        id: check_pr
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # PR to look for
+          PR_TITLE="[Updater Action] Update Notebook and Runtime Images as well as the Commits With New SHAs"
+
+          #Fetch matching PRs
+          PR_NUMBER=$(gh pr list --repo "$REPO_OWNER/$REPO_NAME" --state all --search "$PR_TITLE" --json number,title | jq -r '.[0].number')
+          echo "PR Numbers: $PR_NUMBER"
+
+          if [ -z "$PR_NUMBER" ]; then
+            echo "No PR found with title: $PR_TITLE"
+            exit 1
+          fi
+
+          MAX_ATTEMPTS=30
+          for (( i=1; i<=MAX_ATTEMPTS; i++ )); do
+            echo "Checking if PR #$PR_NUMBER is merged (Attempt $i/$MAX_ATTEMPTS)..."
+            for (( j=1; i<=600; j++ )); do
+              PR_STATE=$(gh pr view --repo "$REPO_OWNER/$REPO_NAME" $PR_NUMBER --json mergedAt --jq '.mergedAt')
+
+              if [ "$PR_STATE" = "null" ] || [ -z "$PR_STATE" ]; then
+                echo "PR #$PR_NUMBER is not merged yet. Waiting..."
+                sleep 1
```

Comment:
Hm, I'm not sure I like the waiting step. At least we should probably sleep for more than 1 second.

Anyway, how do we expect the manifest PR to be merged? Is it gonna be automatically as we discussed last week?

Reply:
> Anyway, how do we expect the manifest PR to be merged? Is it gonna be automatically as we discussed last week?

By hand I figure, same as in Kubeflow repo. Until something more is implemented.

Reply:
Yeah, my expectation was that we plan to wait for the odh on konflux.

Reply:
Quite likely, that's why I was complaining about refining the issue right now, since whatever we decide will be invalidated by the time we end up implementing.

Nevertheless, these concerns don't affect this PR at all, IMO.